app/src/main/kotlin/edu/stanford/bdh/engagehf/health/symptoms/SymptomsUiStateMapper.kt

```diff
@@ -67,40 +46,114 @@ class SymptomsUiStateMapper @Inject constructor() {
         )
     }
 
-    private fun mapTableData(
+    private fun mapNewestHealthData(
         symptomScores: List<SymptomScore>,
         selectedSymptomType: SymptomType,
-    ): List<TableEntryData> {
-        return symptomScores.mapIndexed { index, score ->
-            val previousScore = if (index > 0) symptomScores[index - 1] else null
-            val currentValue = getScoreForSelectedSymptomType(selectedSymptomType, score).toFloat()
-            val previousValue = previousScore?.let {
-                getScoreForSelectedSymptomType(selectedSymptomType, it).toFloat()
-            }
-            val trend = previousValue?.let { currentValue - it } ?: 0f
-            val formattedTrend = if (index > 0 && previousValue != null) {
-                @Suppress("MagicNumber")
-                val trendPercentage = (trend / previousValue) * 100.0f
-                String.format(Locale.getDefault(), PERCENT_FORMAT, trendPercentage)
-            } else {
-                ""
-            }
+    ): NewestHealthData {
+        val newestData = symptomScores.maxByOrNull { it.date }
+        return NewestHealthData(
+            formattedValue = formatValue(newestData, selectedSymptomType),
+            formattedDate = formatHeaderDate(newestData?.date)
+        )
+    }
 
-            TableEntryData(
-                id = null,
-                value = currentValue,
-                secondValue = null,
-                formattedValues = "$currentValue%",
-                date = score.date.toInstant().atZone(ZoneId.systemDefault()),
-                formattedDate = score.date.format(dateTimeFormatter),
-                trend = trend,
-                formattedTrend = formattedTrend
-            )
+    private fun formatValue(newestData: SymptomScore?, selectedSymptomType: SymptomType): String {
+        return when (selectedSymptomType) {
+            SymptomType.OVERALL -> formatOverallScore(newestData)
+            SymptomType.PHYSICAL_LIMITS -> formatPhysicalLimitsScore(newestData)
+            SymptomType.SOCIAL_LIMITS -> formatSocialLimitsScore(newestData)
+            SymptomType.QUALITY_OF_LIFE -> formatQualityOfLifeScore(newestData)
+            SymptomType.SYMPTOMS_FREQUENCY -> formatSymptomsFrequencyScore(newestData)
+            SymptomType.DIZZINESS -> formatDizzinessScore(newestData)
         }
     }
 
+    private fun formatOverallScore(newestData: SymptomScore?): String {
```

Comment:
Can we unify these methods to one?

```kotlin
private fun Double?.asPercent(): String = this?.let { "$it%" } ?: NOT_AVAILABLE
```

and in `formatValue` you can adjust to:

```kotlin
private fun formatValue(newestData: SymptomScore, selectedSymptomType: SymptomType): String {
    return when (selectedSymptomType) {
        SymptomType.OVERALL -> newestData.overallScore.asPercent()
        SymptomType.PHYSICAL_LIMITS -> newestData.physicalLimitsScore.asPercent()
        SymptomType.SOCIAL_LIMITS -> newestData.socialLimitsScore.asPercent()
        SymptomType.QUALITY_OF_LIFE -> newestData.qualityOfLifeScore.asPercent()
        SymptomType.SYMPTOMS_FREQUENCY -> newestData.symptomFrequencyScore.asPercent()
        SymptomType.DIZZINESS -> newestData.dizzinessScore?.toString() ?: NOT_AVAILABLE
    }
}
```

You can also adapt `val newestData = symptomScores.maxBy { it.date }` as it can never be `null` since you check for `isEmpty` before